__tests__/app.test.js

```diff
@@ -102,4 +102,50 @@ describe('/api/articles', ()=>{
             })
         })
     })
+})
+
+describe('/api/articles/:article_id/comments', ()=>{
+    test('GET:200 responds with an array of comments for the given article_id, sorted by date in descending order', ()=>{
+        return request(app).get('/api/articles/9/comments').expect(200)
+        .then(({body: {comments}})=>{
+            expect(comments.length).toBe(2)
+            
+            expect(comments).toBeSortedBy('created_at', {descending: true})
+
+            comments.forEach((comment)=>{
+                expect(comment).toMatchObject(
+                    {
+                        comment_id: expect.any(Number),
+                        votes: expect.any(Number),
+                        article_id: expect.any(Number),
+                        author: expect.any(String),
+                        body: expect.any(String),
+                        created_at: expect.any(String)
+                    }
+                )
+            })
+        })
+    })
+
+    test('GET:200 responds with an empty array when article has no comments', ()=>{
+        return request(app).get('/api/articles/7/comments').expect(200)
+        .then(({body: {comments}})=>{
+            expect(Array.isArray(comments)).toBe(true)
+            expect(comments.length).toBe(0)
+        })
+    })
+
+    test('GET:404 responds with an error message when given a valid but non-existent id', ()=>{
+        return request(app).get('/api/articles/999999/comments').expect(404)
+        .then(({body: {message}})=>{
+            expect(message).toBe('article not found')
+        })
+    })
+
+    test('GET:400 responds with an error message when given an invalid id', ()=>{
+        return request(app).get('/api/articles/dog/comments').expect(400)
+        .then(({body: {message}})=>{
+            expect(message).toBe('Bad request')
+        })
+    })
```

Comment:
Strong tests, well done for remembering that no comments will still be a 200, not an error :)